src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/PreferIsKindAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+            s_localizableMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisPerformance,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: s_localizableDescription,
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                // Kind() methods
+                //  Microsoft.CodeAnalysis.CSharp.CSharpExtensions.Kind
+                //  Microsoft.CodeAnalysis.VisualBasic.VisualBasicExtensions.Kind
+                //
+                // IsKind() methods
+                //  Microsoft.CodeAnalysis.CSharpExtensions.IsKind
+                //  Microsoft.CodeAnalysis.VisualBasicExtensions.IsKind
+
+                Dictionary<INamedTypeSymbol, INamedTypeSymbol> containingTypeMap = new Dictionary<INamedTypeSymbol, INamedTypeSymbol>();
+                if (context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisCSharpCSharpExtensions) is { } csharpKindExtensions
+                    && context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisCSharpExtensions) is { } csharpIsKindExtensions)
+                {
+                    containingTypeMap[csharpKindExtensions] = csharpIsKindExtensions;
+                }
+
+                if (context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisVisualBasicVisualBasicExtensions) is { } vbKindExtensions
+                    && context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisVisualBasicExtensions) is { } vbIsKindExtensions)
+                {
+                    containingTypeMap[vbKindExtensions] = vbIsKindExtensions;
+                }
+
+                if (containingTypeMap.Count > 0)
+                {
+                    context.RegisterOperationAction(context => HandleBinaryOperation(context, containingTypeMap), OperationKind.Binary);
+                }
+            });
+        }
+
+        private static void HandleBinaryOperation(OperationAnalysisContext context, Dictionary<INamedTypeSymbol, INamedTypeSymbol> containingTypeMap)
+        {
+            var operation = (IBinaryOperation)context.Operation;
+            if (operation.OperatorKind is not (BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals))
+            {
+                return;
+            }
+
+            if (operation.LeftOperand.WalkDownConversion() is IInvocationOperation { TargetMethod: { Name: "Kind", ContainingType: var containingType } }
```

Comment:
Consider reporting a diagnostic when the call is on the right side, e.g: `SyntaxKind.IfKeyword == token.Kind()`.

Reply:
enh. :)

Reply:
Agree this case should have been handled.

Reply:
➡️ Decided not to include this one for now because it's a relatively uncommon pattern and it adds substantial complexity to the code fix.

Reply:
> ➡️ Decided not to include this one for now because it's a relatively uncommon pattern and it adds substantial complexity to the code fixes.

Do you want to file an issue that tracks handling this later?